.github/workflows/ci.yml

```diff
@@ -35,27 +35,14 @@ jobs:
     runs-on: ubuntu-latest
     steps:
     - uses: actions/checkout@v3
-    - name: checkout
-      # From https://github.com/actions/checkout/issues/438#issuecomment-1114110354
-      # Required to fetch enough depth for diffs with master in the pre-commit step.
-      # Alternative is to use the checkout action with fetch-depth: 0, but that would
-      # fetch the whole history, potentially slowing down the CI pipeline.
-      run: |
-        commits=${{ github.event.pull_request.commits }}
-        if [[ -n "$commits" ]]; then
-          # Prepare enough depth for diffs with master
-          git fetch --depth="$(( commits + 1 ))"
-        fi
     - uses: actions/setup-python@v3
       with:
           python-version: '3.11'
     - uses: pre-commit/action@v3.0.1
-      with:
-        extra_args: '--from-ref origin/master --to-ref ${{ github.sha }}'
```

Comment:
Let's not drop this part. Instead, let's use `fetch-depth: 0` as described in the comment above. This way, pre-commit will only run on changed files.

This avoids needing to reformat the whole code base, which leads to changes all over the place that are not reviewable because of the sheer mass of changes.

I agree that we can leave out the "fetch optimization", as our history isn't too big, but I wouldn't run the pre-commit hook on the whole codebase but rather just the changed bits.

Reply:
As with this PR all files are fixed so no errors should pop up except if formatting rules change, I usually prefer to run the formatting on all files because if some formatting rule changes, not only newly edited lines follow the new formatting - leading to inconsistent formatting over the whole codebase - but all files are formatted in the same style. Mostly only small changes to formatting rules are updated and those changes are easy to review then. It's just this initial overall formatting which is more extensive. What do you think?

Reply:
Yup sounds good! Would you add those files to the pre-commit ignore command option?

Reply:
Here we go: fdfc554a73ce6beca24ad072df184e10ac3172e6

Reply:
:partying_face:

Reply:
:ship: